### PR TITLE
Add API hook for modifying upserted pages

### PR DESCRIPTION
**Pull request by KyleAMathews**

### CODE REVIEW

**Comment by KyleAMathews** [comment]
To use add something like the following to your `gatsby-node.js`.

```js
exports.onUpsertPage = ({ page, boundActionCreators }) => {
  const { upsertPage } = boundActionCreators
  if (page === SOME_CHECK) {
    // Modify the page e.g.
    page.layout = PATH_TO_DIFFERENT_LAYOUT_COMPONENT
    upsertPage(page)
  }
}
```

File: packages/gatsby/lib/redux/plugin-runner.js [hunk 1]
```diff
@@ -13,4 +13,9 @@ store.subscribe(() => {
     const node = state.nodes[state.lastAction.payload.id]
     apiRunnerNode(`onNodeCreate`, { node })
   }
+
+  if (state.lastAction.type === "UPSERT_PAGE") {
+    const page = state.nodes[state.lastAction.payload]
+    apiRunnerNode(`onUpsertPage`, { page })
```
Comment (jquense) [inline packages/gatsby/lib/redux/plugin-runner.js]: looks like page is undefined here :(

Comment (KyleAMathews) [inline packages/gatsby/lib/redux/plugin-runner.js]: oh... oops. It should just be `state.lastAction.payload`. I'll push a fix for that real quick

Comment (KyleAMathews) [inline packages/gatsby/lib/redux/plugin-runner.js]: Fixed in https://github.com/gatsbyjs/gatsby/commit/ec011f400861e961e537be7e0669c4b5d65aa0e3

Published 1.0.0-alpha13-alpha.ec011f40 w/ fix

**Pull request merged**